explorer/Explorer.tsx

```diff
@@ -341,130 +343,275 @@ export class Explorer
+        ) as string[]
+
+        if (uniqueSlugsInGrapherRow.length) {
+            const baseIndicatorIds = uniq(
+                uniqueSlugsInGrapherRow.flatMap(
+                    this.getBaseIndicatorIdsForColumnWithTransform
+                )
+            )
+                .map((id) => parseInt(id, 10))
+                .filter((id) => !isNaN(id))
+            // add all indicator ids that the given slugs depend on to the config
+            baseIndicatorIds.forEach((indicatorId) => {
                 dimensions.push({
-                    variableId: sizeIndicatorId,
-                    property: DimensionProperty.size,
+                    variableId: indicatorId,
+                    property: DimensionProperty.table, // no specific dimension
                 })
-            }
-            config.dimensions = dimensions
+            })
         }
 
+        config.dimensions = dimensions
+
         grapher.setAuthoredVersion(config)
         grapher.reset()
-        grapher.yAxis.canChangeScaleType = yScaleToggle
-        grapher.yAxis.min = yAxisMin
-        if (facetYDomain) {
-            grapher.yAxis.facetDomain = facetYDomain
-        }
-        if (relatedQuestionText && relatedQuestionUrl) {
-            grapher.relatedQuestions = [
-                { text: relatedQuestionText, url: relatedQuestionUrl },
-            ]
-        }
-        if (mapTargetTime) {
-            grapher.map.time = mapTargetTime
-        }
+        this.updateGrapherFromExplorerCommon()
         grapher.updateFromObject(config)
+        await grapher.downloadLegacyDataFromOwidVariableIds()
+
+        let grapherTable = grapher.inputTable
+
+        // update column definitions with manually provided properties
+        grapherTable = grapherTable.updateDefs((def: OwidColumnDef) => {
+            const manuallyProvidedDef =
+                this.columnDefsNotLinkedToTableByIdOrSlug[def.slug] ?? {}
+            const mergedDef = { ...def, ...manuallyProvidedDef }
+
+            // update display properties
+            mergedDef.display = mergedDef.display ?? {}
+            if (manuallyProvidedDef.name)
+                mergedDef.display.name = manuallyProvidedDef.name
+            if (manuallyProvidedDef.unit)
+                mergedDef.display.unit = manuallyProvidedDef.unit
+            if (manuallyProvidedDef.shortUnit)
+                mergedDef.display.shortUnit = manuallyProvidedDef.shortUnit
+
+            return mergedDef
+        })
 
-        if (
-            creationMode ===
-            ExplorerChartCreationMode.FromExplorerTableColumnSlugs
-        ) {
-            // Clear any error messages, they are likely to be related to dataset loading.
-            this.grapher?.clearErrors()
-            // Set a table immediately. A BlankTable shows a loading animation.
-            this.setGrapherTable(
-                BlankOwidTable(tableSlug, `Loading table '${tableSlug}'`)
+        // add derived columns to grapher table
```

Comment:
It honestly took me way longer than I'd like to admit until I realized why we need to this. This is either a sign of my dwindling attention, or a sign that there should be more of a comment on why we do this, and why we need all this sophisticated dependency tracking.

I hope I'm getting this right:
We need all of that because the transform expects its "source" data columns to all be there in the `OwidTable`, but that's not a given when we construct the table based on indicator ids and the `dimensions` array.
Whereas with csv inputs we would get an `OwidTable` with all columns that are present in the csv file, here we instead get one that is as slim as possible, only containing the variables we directly specify.

However, columns that are source columns for transforms are only indirectly specified (in the transform), and we do all this work to make sure that we detect the dependency on them and also include them in the `dimensions` array, so they are then also part of the table we construct and the transform can run without issues.

Reply:
yes, that's a perfect summary

if spelled out, these are the basic steps:
- given are a bunch of transforms that reference indicators by id or other intermediate columns
- first, figure out which data needs to be downloaded and put those in the `dimensions` array
- after running `downloadLegacyDataFromOwidVariableIds`, the table contains the indicators needed for the transforms (but not the transformed and intermediate columns themselves)
- as a last step, append the transformed and intermediate columns to the input table (this is when the transforms actually run)

that dependency tracking is necessary, like you said, because the price of simply downloading all indicators in the explorers when we encounter a transform would be too high